ui/app/routes/allocations/allocation.js

```diff
@@ -14,18 +14,20 @@ export default Route.extend(WithWatchers, {
   // Allocation breadcrumbs extend from job / task group breadcrumbs
   // even though the route structure does not.
   breadcrumbs(model) {
+    const jobQueryParams = qpBuilder({
+      jobNamespace: model.get('job.namespace.name') || 'default',
+    });
+
     return [
-      { label: 'Jobs', args: ['jobs.index'] },
+      { label: 'Jobs', args: ['jobs.index', jobQueryParams] },
       ...jobCrumbs(model.get('job')),
       {
         label: model.get('taskGroupName'),
         args: [
           'jobs.job.task-group',
-          model.get('job'),
+          model.get('job.plainId'),
```

Comment:
This was `job.name`, which meant the job route was attempting to fetch a job with the wrong ID in the event that name and plainId aren't equal (which is rare but possible).